satellite/gc/piecetracker.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gc
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/bloomfilter"
+	"storj.io/storj/pkg/overlay"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+var (
+	mon = monkit.Package()
+
+	// Error defines the piece tracker errors class
+	Error = errs.Class("piece tracker error")
+)
+
+// RetainInfo contains info needed for a storage node to retain important data and delete garbage data
+type RetainInfo struct {
+	Filter       *bloomfilter.Filter
+	CreationDate time.Time
+	address      *pb.NodeAddress
+	count        int
+}
+
+// PieceTracker allows access to info about the good pieces that storage nodes need to retain
+type PieceTracker interface {
+	// Add adds a RetainInfo to the PieceTracker
```

Comment:
it's not a parameter, but it is created in the method. I'll change the comment though.

Reply:
I saw it but it's something not visible without checking the implementation and the docs shouldn't require to look to the implementation to make them clear.

I found that it could be confused to mention something that it isn't visible.